src/composables/useViewer.js

```diff
@@ -105,13 +105,13 @@ export function useViewer() {
 	})
 
 	const generateViewerObject = (file) => ({
-		fileid: parseInt(file.id, 10),
-		filename: generateAbsolutePath(file.path),
-		basename: file.name,
-		mime: file.mimetype,
-		hasPreview: file.previewAvailable === 'yes' || file['preview-available'] === 'yes',
+		fileid: file.fileid ?? parseInt(file.id, 10),
+		filename: file.filename ?? generateAbsolutePath(file.path),
+		basename: file.basename ?? file.name,
+		mime: file.mime ?? file.mimetype,
+		hasPreview: file.hasPreview ?? (file.previewAvailable === 'yes' || file['preview-available'] === 'yes'),
```

Comment:
Could you describe this change?

Reply:
Object structure differs between file in message.messageParameters and for example, Share API response (mime instead of mimetype, e.g.)
Apart from it, both objects contains permissions as numeric value, and Viewer expects a string like 'RW'

Reply:
If we explicitly have 2 different and specific data structures here, I'd insist on 2 explicit code branches with `if` that constructs Viewer data for the first and for the second case.

Otherwise we potentially have 32 different results here, and it is not clear from the code why we have a backup value for each property.